test/e2e/azure_test.go

```diff
@@ -903,4 +903,55 @@ var _ = Describe("Workload cluster creation", func() {
 			By("PASSED!")
 		})
 	})
+
+	// Workload identity test
+	Context("Creating a cluster that uses workload identity [OPTIONAL]", func() {
```

Comment:
Instead of adding a new test for this, could we switch all the existing tests to use Workload Identity? If we want to keep coverage for Legacy ADD Pod Identity for now we can add an optional test for it.

Reply:
I think the initial plan was to move the functionality to working state and then switch completely in the next PR. Advantage is, this PR is smaller and gets some soak time. 
So we will need to move to using user assigned templates and move to workload identity in the next PR so that it is easy to triage/debug failures if any.
Step by step is what my through process is here. wdyt?